lib/cocoapods/target/build_settings.rb

```diff
@@ -889,12 +900,20 @@ def pod_target_xcconfig_values_by_consumer_by_key
 
         # @return [Array<Sandbox::FileAccessor>]
         define_build_settings_method :file_accessors, :memoized => true do
-          target.file_accessors.select { |fa| fa.spec.spec_type == @xcconfig_spec_type }
+          if non_library_xcconfig?
```

Comment:
can this not combined to a single `select` statement and move that if/else inside?